src/package.cr

```diff
@@ -58,15 +58,20 @@ module Shards
     end
 
     def install
-      cleanup_install_directory
+      Log.with_context do
+        Log.context.set package: name
+        Log.context.set version: report_version
```

Comment:
I added this for debugging purposes. I haven’t identified a use case for regular users yet, but I plan to use it to detect discrepancies between the expected and installed versions.

Let me know if I should keep it or not.

Reply:
Not sure if it's relevant.
But if we keep it, we should merge both into a single context assignment for efficiency.

Also, is there any reason you're using explicit `set` calls instead of passing the data to `with_context`? That'd be more concise.

Reply:
> Also, is there any reason you're using explicit set calls instead of passing the data to with_context? That'd be more concise.

Good call - because the `with_context(**kwargs)` was introduced after crystal 1.0.0. I found there are still tests are runing against Crystal 1.0.0 that are failing. [matrix](https://github.com/crystal-lang/shards/blob/master/.github/workflows/ci.yml#L19-L20)

I thought to introduce patch for older version to support `with_context(**kwargs)`.

https://github.com/crystal-lang/crystal/pull/11517

Reply:
Okay, that's a good reason 😆 
I like the idea of adding a polyfill for Crystal < 1.3.
However, it would be fine to keep a separate `set` call either.